File: javascript_version/public/javascripts/lib/Behavior.js
var Behavior = function(name, key, description){
  this.name = name;
  this.key = key.toLowerCase();
  this.description = description;
  this.frequency = 0;
  this.rate = null;

  this.increment = function(amount){
    if(!amount) { amount = 1; }
    this.frequency += amount;
  };
};

File: console_version/lib/behavior.rb
class Behavior
  attr_reader :name, :key, :description, :frequency

  def initialize(name, key, description = nil)
    @name = name
    @key = key
    @description = description
    @frequency = 0
  end

  def increment(num = 1)
    @frequency += num
  end
end
